### PR TITLE
Refactor/cleanup/reduce slow tests

### PR DESCRIPTION
**Pull request by stefanpenner**
- [x] addon-smoke test 70s -> 19s on my laptop.  (more can be done)
- [x] remove unused fixtures

### CODE REVIEW

File: tests/runner.js [hunk 1]
```diff
@@ -10,8 +10,8 @@ if (process.env.EOLNEWLINE) {
   require('os').EOL = '\n';
 }
 
-fs.removeSync('.node_modules-tmp');
-fs.removeSync('.bower_components-tmp');
+// fs.removeSync('.node_modules-tmp');
```
Comment (stefanpenner) [inline tests/runner.js]: NOTE: remove before merging

File: tests/acceptance/addon-smoke-test-slow.js [hunk 2]
```diff
@@ -22,14 +22,16 @@ var chai = require('../chai');
...
+var addonName = 'some-cool-addon';
+var addonRoot = path.resolve(__dirname + '/../../tmp/' + addonName) + '/';
 
 describe('Acceptance: addon-smoke-test', function() {
   this.timeout(450000);
 
   before(function() {
     return createTestTargets(addonName, {
       command: 'addon'
+    }).finally(function() {
```
Comment (twokul) [inline tests/acceptance/addon-smoke-test-slow.js]: empty finally block?

Comment (stefanpenner) [inline tests/acceptance/addon-smoke-test-slow.js]: ah, most likely from logging. Will remove, thanks.

**Comment by nathanhammond** [comment]
@stefanpenner you've got some weird git history with `af57a6b` being included in this PR.

File: tests/acceptance/addon-smoke-test-slow.js [hunk 3]
```diff
@@ -64,137 +67,47 @@ describe('Acceptance: addon-smoke-test', function() {
...
-              killCliProcess(child);
-            }
-          }
-        })
-        .catch(function() {
-          // just eat the rejection as we are testing what happens
-        });
-      });
-  });
+  it('kitchen sink', function() {
```
Comment (nathanhammond) [inline tests/acceptance/addon-smoke-test-slow.js]: Can we get a better description?

Comment (nathanhammond) [inline tests/acceptance/addon-smoke-test-slow.js]: (I understand it, but it'll help people who don't have that colloquialism.)

File: tests/acceptance/addon-smoke-test-slow.js [hunk 4]
```diff
@@ -250,41 +163,4 @@ describe('Acceptance: addon-smoke-test', function() {
...
-    return new Promise(function(resolve) {
-      setTimeout(function() {
-        resolve();
-      }, time);
-    });
-  }
-
-  var isWindows = /^win/.test(process.platform);
-
-  (isWindows ? it.skip : it)('doesn\'t fail to build new files', function() {
```
Comment (nathanhammond) [inline tests/acceptance/addon-smoke-test-slow.js]: What's the story about the bug that made us add this test? How likely is it to reoccur? (It looks like we're testing across the testem boundary here. Maybe @johanneswuerbach should be aware that we're not doing this anymore...)

Comment (stefanpenner) [inline tests/acceptance/addon-smoke-test-slow.js]: I removed this, because  it provides little value and high cost.

File: tests/acceptance/addon-smoke-test-slow.js (hunk 3 above)
Comment (stefanpenner) [inline tests/acceptance/addon-smoke-test-slow.js]: sure, any suggestions?

**Comment by nathanhammond** [comment]
@stefanpenner for the brocfile tests is that just an acknowledgment that those tests are covered by other scenarios?

I'm generally 👍 on merging, but it feels _really_ strange to delete tests.

**Comment by stefanpenner** [comment]
> @stefanpenner for the brocfile tests is that just an acknowledgment that those tests are covered by other scenarios?

ya

File: tests/acceptance/addon-smoke-test-slow.js (hunk 4 above)
Comment (johanneswuerbach) [inline tests/acceptance/addon-smoke-test-slow.js]: Agreed, I'll also take an addional look if this is test covered within testem.

File: tests/acceptance/addon-smoke-test-slow.js (hunk 3 above)
Comment (nathanhammond) [inline tests/acceptance/addon-smoke-test-slow.js]: `works in most common scenarios for an example application`

**Comment by nathanhammond** [comment]
Land the description change and I'm `r+`.

**Comment by nathanhammond** [comment]
Oh hey, it's on this repo because it's @stefanpenner! Handling it.

**Comment by nathanhammond** [comment]
@homu r+
